firmware/lib/Comm/src/ArduinoSerialDevice.cxx

```diff
@@ -27,9 +27,10 @@ uint32_t ArduinoSerialDevice::ser_available()
     return this->device.available();
 }
 
-uint8_t ArduinoSerialDevice::ser_read()
+bool ArduinoSerialDevice::ser_read(uint8_t *out)
 {
-    return this->device.read();
+    *out = this->device.read();
+    return true;
```

Comment:
Does this always return true?

Reply:
Intentionally always returns true. On Arduino there's no way to tell if the read was successful or not, so we have to assume it was